### ISSUE TITLE
aws_s3_bucket fails in custom environment lacking Bucket Object Lock Configuration

### ISSUE THREAD

**Issue by anowell** [terraform-providers/terraform-provider-aws#10153]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

terraform 0.11.8
terraform-aws 2.26.0

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

```hcl
resource "aws_s3_bucket" "main" {
  bucket        = "some_unique_bucket_name"
  force_destroy = true
}
```

### Expected Behavior

Should create the S3 bucket as expected.

### Actual Behavior

Error in custom environment:

```
* aws_s3_bucket.main: aws_s3_bucket.main: error getting S3 Bucket Object Lock configuration: MethodNotAllowed: The specified method is not allowed against this resource.
    status code: 405, request id: ...snip..., host id: ...snip...
```

### Steps to Reproduce

Requires environment that does not support Bucket Object Lock Configuration. Then:

1. `terraform apply`

### Important Factoids

This only occurs in environments without Bucket Object Lock Configuration support. This is not an issue for commercial AWS.

### References

I am fairly confident c6c1dc0 in v1.56 is the addition that breaks in this environment when attempting a `resourceAwsS3BucketRead`. It seems to me that [this error handling](https://github.com/terraform-providers/terraform-provider-aws/blob/a02170ed8f37ac610b25ef883b542882f9b92ffa/aws/resource_aws_s3_bucket.go#L2450) should treat `MethodNotAllowed` the same as `ObjectLockConfigurationNotFoundError`. I'm not yet sure if that change would be fully sufficient, but I am intending to build a patched provider with this change to confirm, and would happily submit a PR. We currently depend on other fixes in the provider that prevent using versions <2.4, so I'm motivated to help support this scenario, and any guidance would be welcome.

**Comment by bflad** [comment]
Hi @anowell 👋 Your fix proposal looks correct seeing the error message -- adding `MethodNotAllowed` handling there should bypass this issue, e.g.

```go
	// Certain S3 implementations do not include this API
	if isAWSErr(err, "MethodNotAllowed", "") {
		return nil, nil
	}

	if isAWSErr(err, "ObjectLockConfigurationNotFoundError", "") {
		return nil, nil
	}

	if err != nil {
		return nil, err
	}
```

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.34.0 of the Terraform AWS Provider, on Thursday. Thanks for contributing the change, @anowell 👍

**Comment by bflad** [comment]
This has been released in [version 2.34.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2340-october-31-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
